fix: updated icon button to include disabled flag for repeatable comp

ankit7201 opened the pull request:
<!--
Hello 👋 Thank you for submitting a pull request.

To help us merge your PR, make sure to follow the instructions below:

- Create or update the tests
- Create or update the documentation at https://github.com/strapi/documentation
- Refer to the issue you are closing in the PR description: Fix #issue
- Specify if the PR is ready to be merged or work in progress (by opening a draft PR)

Please ensure you read the Contributing Guide: https://github.com/strapi/strapi/blob/main/CONTRIBUTING.md
-->

### What does it do?

Disables the icon button for repeatable component in published tab 

### Why is it needed?

See #23305 

### How to test it?

Go to a repeatable component and try clicking on delete icon when in Published tab. 
Nothing will happen as it is disabled. 

Go to draft tab and click the delete button and it will work. 

### Related issue(s)/PR(s)

#23305

ankit7201 commented:
Hi @lucasboilly 

Can you please review this?

lucasboilly commented:
Hi @ankit7201, really cool! 

Could you disable the drag `IconButton` that's just next to it as well? And add a 4px gap between the 2 `IconButtons` so they're not sticked together?

Thanks for your help 🙌

ankit7201 commented:
Hi @lucasboilly 

I have updated it and now drag button is also disabled. 

Regarding the 4px gap task, I have added the condition to add the gap only is disabled mode since in non disabled mode, the button spacing looks good. 
Please correct me if I am wrong, but wouldn't it make more sense to fix the gap issue in strapi design system (as a separate PR) rather than here since this seems like a general issue and accordion should control how its icons should look like.

lucasboilly commented:
@ankit7201 you are right about the gap, let's forget this and we'll fix it later in the DS 👍

So you just need to remove the condition and the `Flex` around the buttons if I'm not wrong?

It's all good regarding the `Trash` and `Drag` icon buttons 👌

ankit7201 commented:
@lucasboilly 
Reverted the gap code. 
Thanks.

lucasboilly commented:
All good, thanks again @ankit7201!

I'm gonna share it to our engineers for the code review :)

lucasboilly commented:
Hi @ankit7201, your code was shipped in the last Strapi version ([5.12.7](https://github.com/strapi/strapi/releases/tag/v5.12.7)) :)

ankit7201 commented:
@lucasboilly 
Awesome. Thanks.
